src/main/kotlin/com/manimdsl/stylesheet/Stylesheet.kt

```diff
@@ -0,0 +1,93 @@
+package com.manimdsl.stylesheet
+
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.gson.reflect.TypeToken
+import com.manimdsl.errorhandling.ErrorHandler
+import com.manimdsl.errorhandling.warnings.undeclaredVariableStyleWarning
+import com.manimdsl.executor.ExecValue
+import com.manimdsl.frontend.SymbolTableVisitor
+import java.io.File
+import java.lang.reflect.Type
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.primaryConstructor
+import kotlin.system.exitProcess
+
+interface StylesheetProperty {
+    val borderColor: String?
+    val textColor: String?
+}
+
+data class AnimationProperties(override val borderColor: String? = null, override val textColor: String? = null) :
+    StylesheetProperty
+
+data class StyleProperties(
+    override val borderColor: String? = null,
+    override val textColor: String? = null,
+    val animate: AnimationProperties? = null
+) : StylesheetProperty
+
+
+class Stylesheet(private val stylesheetPath: String?, private val symbolTableVisitor: SymbolTableVisitor) {
+
+    private val stylesheet: Map<String, StyleProperties>
+    private val dataStructureStrings = setOf("Stack")
+
+    init {
+        stylesheet = if (stylesheetPath != null) {
+            parseStylesheet(stylesheetPath)
+        } else {
+            emptyMap()
+        }
+    }
+
+    private fun parseStylesheet(stylesheetPath: String): Map<String, StyleProperties> {
+        val gson = Gson()
+        val type: Type = object : TypeToken<Map<String, StyleProperties>>() {}.type
+        return try {
+            val stylesheetMap: Map<String, StyleProperties> = gson.fromJson(File(stylesheetPath).readText(), type)
+            stylesheetMap.keys.forEach {
+                if (!(dataStructureStrings.contains(it) || symbolTableVisitor.getVariableNames().contains(it))) {
+                    undeclaredVariableStyleWarning(it)
+                }
+            }
+            ErrorHandler.checkWarnings()
```

Comment:
I see that there is an `ErrorHandler.checkErrorsAndWarnings()`. If later we add warnings to previous stages i.e. lexical, syntax or semantic analysis then these warnings will be printed twice as `ErrorHandler.checkErrorsAndWarnings()` is called before checkWarnings and warnings are non-terminating.

Reply:
I'm not sure because I think Rini clears the warnings after printing